### ISSUE TITLE
libev is not included in libwebsockets.a / building with libev fails

### ISSUE THREAD

**Issue by alexhultman** [warmcat/libwebsockets#546]
Hi,

I'm trying to compile libwebsockets and I use this cmake configuration:

`cmake -DLWS_USE_LIBEV=1 .`

Building the lib works, but the test apps fail to link due to missing libev.c in the compilation. If I skip building the test apps, it succeeds but the app I'm building fails due to the same undefined references. If I explicitly link to libwebsockets and also libev.c it works.

**Comment by lws-team** [comment]
Can't reproduce on master... travis tests for this too and has not had a problem now or in the last weeks at least.  Eg,

https://travis-ci.org/warmcat/libwebsockets/jobs/134328173

**Comment by alexhultman** [comment]
Yeah well I have and can reproduce it on two separate computers on a cleanly cloned repo:

```
[alexhultman@localhost ~]$ git clone https://github.com/warmcat/libwebsockets.git
Cloning into 'libwebsockets'...
remote: Counting objects: 10771, done.
remote: Compressing objects: 100% (27/27), done.
remote: Total 10771 (delta 12), reused 0 (delta 0), pack-reused 10744
Receiving objects: 100% (10771/10771), 8.00 MiB | 2.96 MiB/s, done.
Resolving deltas: 100% (8096/8096), done.
Checking connectivity... done.
[alexhultman@localhost ~]$ cd libwebsockets/
[alexhultman@localhost libwebsockets]$ cmake -DLWS_USE_LIBEV=1 .
-- The C compiler identification is GNU 5.3.1
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- CMAKE_TOOLCHAIN_FILE=''
-- Found Git: /usr/bin/git  
Git commit hash: alexhultman@localhost.localdomain-v2.0.0-61-g3d71eae
-- Performing Test LWS_HAVE_inline
-- Performing Test LWS_HAVE_inline - Success
-- Performing Test LWS_HAVE___inline__
-- Performing Test LWS_HAVE___inline__ - Success
-- Performing Test LWS_HAVE___inline
-- Performing Test LWS_HAVE___inline - Success
-- Looking for bzero
-- Looking for bzero - found
-- Looking for fork
-- Looking for fork - found
-- Looking for getenv
-- Looking for getenv - found
-- Looking for malloc
-- Looking for malloc - found
-- Looking for memset
-- Looking for memset - found
-- Looking for realloc
-- Looking for realloc - found
-- Looking for socket
-- Looking for socket - found
-- Looking for strerror
-- Looking for strerror - found
-- Looking for vfork
-- Looking for vfork - found
-- Looking for execvpe
-- Looking for execvpe - found
-- Looking for getifaddrs
-- Looking for getifaddrs - found
-- Looking for snprintf
-- Looking for snprintf - found
-- Looking for _snprintf
-- Looking for _snprintf - not found
-- Looking for _vsnprintf
-- Looking for _vsnprintf - not found
-- Looking for getloadavg
-- Looking for getloadavg - found
-- Looking for dlfcn.h
-- Looking for dlfcn.h - found
-- Looking for fcntl.h
-- Looking for fcntl.h - found
-- Looking for in6addr.h
-- Looking for in6addr.h - not found
-- Looking for inttypes.h
-- Looking for inttypes.h - found
-- Looking for memory.h
-- Looking for memory.h - found
-- Looking for netinet/in.h
-- Looking for netinet/in.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stdlib.h
-- Looking for stdlib.h - found
-- Looking for strings.h
-- Looking for strings.h - found
-- Looking for string.h
-- Looking for string.h - found
-- Looking for sys/prctl.h
-- Looking for sys/prctl.h - found
-- Looking for sys/socket.h
-- Looking for sys/socket.h - found
-- Looking for sys/stat.h
-- Looking for sys/stat.h - found
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for unistd.h
-- Looking for unistd.h - found
-- Looking for vfork.h
-- Looking for vfork.h - not found
-- Looking for zlib.h
-- Looking for zlib.h - found
-- Looking for 4 include files stdlib.h, ..., float.h
-- Looking for 4 include files stdlib.h, ..., float.h - found
-- Performing Test LWS_HAVE_VISIBILITY
-- Performing Test LWS_HAVE_VISIBILITY - Success
-- Found ZLIB: /usr/lib64/libz.so (found version "1.2.8") 
zlib include dirs: /usr/include
zlib libraries: /usr/lib64/libz.so
Compiling with SSL support
-- Found OpenSSL: /usr/lib64/libssl.so;/usr/lib64/libcrypto.so (found version "1.0.1k") 
OpenSSL include dir: /usr/include
OpenSSL libraries: /usr/lib64/libssl.so;/usr/lib64/libcrypto.so
-- Looking for openssl/ecdh.h
-- Looking for openssl/ecdh.h - found
-- Looking for SSL_CTX_set1_param
-- Looking for SSL_CTX_set1_param - found
Searching for OpenSSL executable and dlls
OpenSSL executable: /usr/bin/openssl
Generating SSL Certificates for the test-server...
SUCCESSFULLY generated SSL certificate
Generating API documentation
-- Looking for RPMTools... - found rpmuild is /usr/bin/rpmbuild
-- RPMTools:: Using RPM_ROOTDIR=/home/alexhultman/libwebsockets/RPM
-- Simple copy spec file <libwebsockets.spec> --> </home/alexhultman/libwebsockets/RPM/SPECS/libwebsockets.spec>
---------------------------------------------------------------------
  Settings:  (For more help do cmake -LH <srcpath>)
---------------------------------------------------------------------
 LWS_WITH_STATIC = ON
 LWS_WITH_SHARED = ON
 LWS_WITH_SSL = ON (SSL Support)
 LWS_SSL_CLIENT_USE_OS_CA_CERTS = 1
 LWS_USE_WOLFSSL = OFF (wolfSSL/CyaSSL replacement for OpenSSL)
 LWS_USE_POLARSSL = OFF (PolarSSL replacement for OpenSSL)
 LWS_USE_MBEDTLS = OFF (mbedtls (nee Polarssl) replacement for OpenSSL)
 LWS_WITHOUT_BUILTIN_SHA1 = OFF
 LWS_WITHOUT_BUILTIN_GETIFADDRS = OFF
 LWS_WITHOUT_CLIENT = OFF
 LWS_WITHOUT_SERVER = OFF
 LWS_LINK_TESTAPPS_DYNAMIC = OFF
 LWS_WITHOUT_TESTAPPS = OFF
 LWS_WITHOUT_TEST_SERVER = OFF
 LWS_WITHOUT_TEST_SERVER_EXTPOLL = OFF
 LWS_WITHOUT_TEST_PING = OFF
 LWS_WITHOUT_TEST_ECHO = OFF
 LWS_WITHOUT_TEST_CLIENT = OFF
 LWS_WITHOUT_TEST_FRAGGLE = OFF
 LWS_WITHOUT_EXTENSIONS = OFF
 LWS_WITH_LATENCY = OFF
 LWS_WITHOUT_DAEMONIZE = ON
 LWS_USE_LIBEV = 1
 LWS_USE_LIBUV = 
 LWS_IPV6 = OFF
 LWS_UNIX_SOCK = OFF
 LWS_WITH_HTTP2 = OFF
 LWS_MBED3 = OFF
 LWS_SSL_SERVER_WITH_ECDH_CERT = OFF
 LWS_MAX_SMP = 32
 LWS_WITH_CGI = OFF
 LWS_HAVE_OPENSSL_ECDH_H = 1
 LWS_HAVE_SSL_CTX_set1_param = 1
 LWS_WITH_HTTP_PROXY = OFF
 LIBHUBBUB_LIBRARIES = 
 PLUGINS = 
 LWS_WITH_ACCESS_LOG = OFF
 LWS_WITH_SERVER_STATUS = OFF
 LWS_WITH_LEJP = OFF
 LWS_WITH_LEJP_CONF = OFF
 LWS_WITH_GENERIC_SESSIONS = OFF
---------------------------------------------------------------------
-- Configuring done
-- Generating done
-- Build files have been written to: /home/alexhultman/libwebsockets
[alexhultman@localhost libwebsockets]$ make
Scanning dependencies of target websockets
[  1%] Building C object CMakeFiles/websockets.dir/lib/base64-decode.c.o
[  2%] Building C object CMakeFiles/websockets.dir/lib/handshake.c.o
[  3%] Building C object CMakeFiles/websockets.dir/lib/libwebsockets.c.o
[  5%] Building C object CMakeFiles/websockets.dir/lib/service.c.o
[  6%] Building C object CMakeFiles/websockets.dir/lib/pollfd.c.o
[  7%] Building C object CMakeFiles/websockets.dir/lib/output.c.o
[  9%] Building C object CMakeFiles/websockets.dir/lib/parsers.c.o
[ 10%] Building C object CMakeFiles/websockets.dir/lib/context.c.o
[ 11%] Building C object CMakeFiles/websockets.dir/lib/alloc.c.o
[ 12%] Building C object CMakeFiles/websockets.dir/lib/header.c.o
[ 14%] Building C object CMakeFiles/websockets.dir/lib/client.c.o
[ 15%] Building C object CMakeFiles/websockets.dir/lib/client-handshake.c.o
[ 16%] Building C object CMakeFiles/websockets.dir/lib/client-parser.c.o
[ 18%] Building C object CMakeFiles/websockets.dir/lib/ssl.c.o
[ 19%] Building C object CMakeFiles/websockets.dir/lib/ssl-server.c.o
[ 20%] Building C object CMakeFiles/websockets.dir/lib/ssl-client.c.o
[ 22%] Building C object CMakeFiles/websockets.dir/lib/sha-1.c.o
[ 23%] Building C object CMakeFiles/websockets.dir/lib/lws-plat-unix.c.o
[ 24%] Building C object CMakeFiles/websockets.dir/lib/server.c.o
[ 25%] Building C object CMakeFiles/websockets.dir/lib/server-handshake.c.o
[ 27%] Building C object CMakeFiles/websockets.dir/lib/extension.c.o
[ 28%] Building C object CMakeFiles/websockets.dir/lib/extension-permessage-deflate.c.o
[ 29%] Linking C static library lib/libwebsockets.a
[ 29%] Built target websockets
Scanning dependencies of target test-client
[ 31%] Building C object CMakeFiles/test-client.dir/test-server/test-client.c.o
[ 32%] Linking C executable bin/libwebsockets-test-client
lib/libwebsockets.a(pollfd.c.o): In function `_lws_change_pollfd':
pollfd.c:(.text+0x159): undefined reference to `lws_libev_io'
pollfd.c:(.text+0x169): undefined reference to `lws_libev_io'
pollfd.c:(.text+0x181): undefined reference to `lws_libev_io'
pollfd.c:(.text+0x199): undefined reference to `lws_libev_io'
lib/libwebsockets.a(pollfd.c.o): In function `remove_wsi_socket_from_fds':
pollfd.c:(.text+0x498): undefined reference to `lws_libev_io'
lib/libwebsockets.a(pollfd.c.o):pollfd.c:(.text+0x866): more undefined references to `lws_libev_io' follow
lib/libwebsockets.a(context.c.o): In function `lws_context_destroy':
context.c:(.text+0x8f7): undefined reference to `lws_libev_destroyloop'
lib/libwebsockets.a(context.c.o): In function `lws_create_context':
context.c:(.text+0xb24): undefined reference to `lws_feature_status_libev'
context.c:(.text+0xd7c): undefined reference to `lws_ev_sigint_cb'
lib/libwebsockets.a(client-handshake.c.o): In function `lws_client_connect_2':
client-handshake.c:(.text+0x434): undefined reference to `lws_libev_accept'
lib/libwebsockets.a(lws-plat-unix.c.o): In function `lws_plat_service_tsi':
lws-plat-unix.c:(.text+0x2e5): undefined reference to `lws_libev_run'
lib/libwebsockets.a(lws-plat-unix.c.o): In function `lws_plat_insert_socket_into_fds':
lws-plat-unix.c:(.text+0x8b9): undefined reference to `lws_libev_io'
lib/libwebsockets.a(lws-plat-unix.c.o): In function `lws_plat_delete_socket_from_fds':
lws-plat-unix.c:(.text+0x909): undefined reference to `lws_libev_io'
lib/libwebsockets.a(lws-plat-unix.c.o): In function `lws_plat_init':
lws-plat-unix.c:(.text+0x9fb): undefined reference to `lws_libev_init_fd_table'
lib/libwebsockets.a(server.c.o): In function `lws_adopt_socket_vhost':
server.c:(.text+0x60b): undefined reference to `lws_libev_accept'
lib/libwebsockets.a(server.c.o): In function `lws_adopt_socket':
server.c:(.text+0x7e2): undefined reference to `lws_libev_accept'
collect2: error: ld returned 1 exit status
CMakeFiles/test-client.dir/build.make:98: recipe for target 'bin/libwebsockets-test-client' failed
make[2]: *** [bin/libwebsockets-test-client] Error 1
CMakeFiles/Makefile2:163: recipe for target 'CMakeFiles/test-client.dir/all' failed
make[1]: *** [CMakeFiles/test-client.dir/all] Error 2
Makefile:149: recipe for target 'all' failed
make: *** [all] Error 2
[alexhultman@localhost libwebsockets]$
```

**Comment by lws-team** [comment]
Here is what I tested before posting about it

```
[agreen@build build5]$ stg branch master
Checking for changes in the working directory ... done
Switching to branch "master" ... done
[agreen@build build5]$ mkdir ../build8
[agreen@build build5]$ cd ../build8
[agreen@build build8]$ cmake .. -DLWS_WITH_LIBEV=1
-- The C compiler identification is GNU 6.1.1
-- Check for working C compiler: /usr/lib64/ccache/cc
-- Check for working C compiler: /usr/lib64/ccache/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- CMAKE_TOOLCHAIN_FILE=''
-- Found Git: /usr/bin/git  
Git commit hash: agreen@build.warmcat.com-v2.0.0-61-g4b86352
-- Performing Test LWS_HAVE_inline
-- Performing Test LWS_HAVE_inline - Success
-- Performing Test LWS_HAVE___inline__
-- Performing Test LWS_HAVE___inline__ - Success
-- Performing Test LWS_HAVE___inline
-- Performing Test LWS_HAVE___inline - Success
-- Looking for bzero
-- Looking for bzero - found
-- Looking for fork
-- Looking for fork - found
-- Looking for getenv
-- Looking for getenv - found
-- Looking for malloc
-- Looking for malloc - found
-- Looking for memset
-- Looking for memset - found
-- Looking for realloc
-- Looking for realloc - found
-- Looking for socket
-- Looking for socket - found
-- Looking for strerror
-- Looking for strerror - found
-- Looking for vfork
-- Looking for vfork - found
-- Looking for execvpe
-- Looking for execvpe - found
-- Looking for getifaddrs
-- Looking for getifaddrs - found
-- Looking for snprintf
-- Looking for snprintf - found
-- Looking for _snprintf
-- Looking for _snprintf - not found
-- Looking for _vsnprintf
-- Looking for _vsnprintf - not found
-- Looking for getloadavg
-- Looking for getloadavg - found
-- Looking for dlfcn.h
-- Looking for dlfcn.h - found
-- Looking for fcntl.h
-- Looking for fcntl.h - found
-- Looking for in6addr.h
-- Looking for in6addr.h - not found
-- Looking for inttypes.h
-- Looking for inttypes.h - found
-- Looking for memory.h
-- Looking for memory.h - found
-- Looking for netinet/in.h
-- Looking for netinet/in.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stdlib.h
-- Looking for stdlib.h - found
-- Looking for strings.h
-- Looking for strings.h - found
-- Looking for string.h
-- Looking for string.h - found
-- Looking for sys/prctl.h
-- Looking for sys/prctl.h - found
-- Looking for sys/socket.h
-- Looking for sys/socket.h - found
-- Looking for sys/stat.h
-- Looking for sys/stat.h - found
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for unistd.h
-- Looking for unistd.h - found
-- Looking for vfork.h
-- Looking for vfork.h - not found
-- Looking for zlib.h
-- Looking for zlib.h - found
-- Looking for 4 include files stdlib.h, ..., float.h
-- Looking for 4 include files stdlib.h, ..., float.h - found
-- Performing Test LWS_HAVE_VISIBILITY
-- Performing Test LWS_HAVE_VISIBILITY - Success
-- Found ZLIB: /usr/lib64/libz.so (found version "1.2.8") 
zlib include dirs: /usr/include
zlib libraries: /usr/lib64/libz.so
Compiling with SSL support
-- Found OpenSSL: /usr/lib64/libssl.so;/usr/lib64/libcrypto.so (found version "1.0.2h") 
OpenSSL include dir: /usr/include
OpenSSL libraries: /usr/lib64/libssl.so;/usr/lib64/libcrypto.so
-- Looking for openssl/ecdh.h
-- Looking for openssl/ecdh.h - found
libev include dir: /usr/include
libev libraries: /usr/lib64/libev.so
-- Looking for SSL_CTX_set1_param
-- Looking for SSL_CTX_set1_param - found
Searching for OpenSSL executable and dlls
OpenSSL executable: /usr/bin/openssl
Generating SSL Certificates for the test-server...
SUCCESSFULLY generated SSL certificate
Generating API documentation
-- Looking for RPMTools... - found rpmuild is /usr/bin/rpmbuild
-- RPMTools:: Using RPM_ROOTDIR=/projects/libwebsockets/build8/RPM
-- Simple copy spec file <libwebsockets.spec> --> </projects/libwebsockets/build8/RPM/SPECS/libwebsockets.spec>
---------------------------------------------------------------------
  Settings:  (For more help do cmake -LH <srcpath>)
---------------------------------------------------------------------
 LWS_WITH_STATIC = ON
 LWS_WITH_SHARED = ON
 LWS_WITH_SSL = ON (SSL Support)
 LWS_SSL_CLIENT_USE_OS_CA_CERTS = 1
 LWS_USE_WOLFSSL = OFF (wolfSSL/CyaSSL replacement for OpenSSL)
 LWS_USE_POLARSSL = OFF (PolarSSL replacement for OpenSSL)
 LWS_USE_MBEDTLS = OFF (mbedtls (nee Polarssl) replacement for OpenSSL)
 LWS_WITHOUT_BUILTIN_SHA1 = OFF
 LWS_WITHOUT_BUILTIN_GETIFADDRS = OFF
 LWS_WITHOUT_CLIENT = OFF
 LWS_WITHOUT_SERVER = OFF
 LWS_LINK_TESTAPPS_DYNAMIC = OFF
 LWS_WITHOUT_TESTAPPS = OFF
 LWS_WITHOUT_TEST_SERVER = OFF
 LWS_WITHOUT_TEST_SERVER_EXTPOLL = OFF
 LWS_WITHOUT_TEST_PING = OFF
 LWS_WITHOUT_TEST_ECHO = OFF
 LWS_WITHOUT_TEST_CLIENT = OFF
 LWS_WITHOUT_TEST_FRAGGLE = OFF
 LWS_WITHOUT_EXTENSIONS = OFF
 LWS_WITH_LATENCY = OFF
 LWS_WITHOUT_DAEMONIZE = ON
 LWS_USE_LIBEV = 1
 LWS_USE_LIBUV = 
 LWS_IPV6 = OFF
 LWS_UNIX_SOCK = OFF
 LWS_WITH_HTTP2 = OFF
 LWS_MBED3 = OFF
 LWS_SSL_SERVER_WITH_ECDH_CERT = OFF
 LWS_MAX_SMP = 32
 LWS_WITH_CGI = OFF
 LWS_HAVE_OPENSSL_ECDH_H = 1
 LWS_HAVE_SSL_CTX_set1_param = 1
 LWS_WITH_HTTP_PROXY = OFF
 LIBHUBBUB_LIBRARIES = 
 PLUGINS = 
 LWS_WITH_ACCESS_LOG = OFF
 LWS_WITH_SERVER_STATUS = OFF
 LWS_WITH_LEJP = OFF
 LWS_WITH_LEJP_CONF = OFF
 LWS_WITH_GENERIC_SESSIONS = OFF
---------------------------------------------------------------------
-- Configuring done
-- Generating done
-- Build files have been written to: /projects/libwebsockets/build8
[agreen@build build8]$ make && sudo make install && sudo lwsws
Scanning dependencies of target websockets
[  1%] Building C object CMakeFiles/websockets.dir/lib/base64-decode.c.o
[  2%] Building C object CMakeFiles/websockets.dir/lib/handshake.c.o
[  3%] Building C object CMakeFiles/websockets.dir/lib/libwebsockets.c.o
[  4%] Building C object CMakeFiles/websockets.dir/lib/service.c.o
[  5%] Building C object CMakeFiles/websockets.dir/lib/pollfd.c.o
[  6%] Building C object CMakeFiles/websockets.dir/lib/output.c.o
[  8%] Building C object CMakeFiles/websockets.dir/lib/parsers.c.o
[  9%] Building C object CMakeFiles/websockets.dir/lib/context.c.o
[ 10%] Building C object CMakeFiles/websockets.dir/lib/alloc.c.o
[ 11%] Building C object CMakeFiles/websockets.dir/lib/header.c.o
[ 12%] Building C object CMakeFiles/websockets.dir/lib/client.c.o
[ 13%] Building C object CMakeFiles/websockets.dir/lib/client-handshake.c.o
[ 15%] Building C object CMakeFiles/websockets.dir/lib/client-parser.c.o
[ 16%] Building C object CMakeFiles/websockets.dir/lib/ssl.c.o
[ 17%] Building C object CMakeFiles/websockets.dir/lib/ssl-server.c.o
[ 18%] Building C object CMakeFiles/websockets.dir/lib/ssl-client.c.o
[ 19%] Building C object CMakeFiles/websockets.dir/lib/sha-1.c.o
[ 20%] Building C object CMakeFiles/websockets.dir/lib/lws-plat-unix.c.o
[ 22%] Building C object CMakeFiles/websockets.dir/lib/server.c.o
[ 23%] Building C object CMakeFiles/websockets.dir/lib/server-handshake.c.o
[ 24%] Building C object CMakeFiles/websockets.dir/lib/extension.c.o
[ 25%] Building C object CMakeFiles/websockets.dir/lib/extension-permessage-deflate.c.o
[ 26%] Building C object CMakeFiles/websockets.dir/lib/libev.c.o
[ 27%] Linking C static library lib/libwebsockets.a
[ 27%] Built target websockets
Scanning dependencies of target test-fuzxy
[ 29%] Building C object CMakeFiles/test-fuzxy.dir/test-server/fuzxy.c.o
[ 30%] Linking C executable bin/libwebsockets-test-fuzxy
[ 30%] Built target test-fuzxy
Scanning dependencies of target websockets_shared
[ 31%] Building C object CMakeFiles/websockets_shared.dir/lib/base64-decode.c.o
[ 32%] Building C object CMakeFiles/websockets_shared.dir/lib/handshake.c.o
[ 33%] Building C object CMakeFiles/websockets_shared.dir/lib/libwebsockets.c.o
[ 34%] Building C object CMakeFiles/websockets_shared.dir/lib/service.c.o
[ 36%] Building C object CMakeFiles/websockets_shared.dir/lib/pollfd.c.o
[ 37%] Building C object CMakeFiles/websockets_shared.dir/lib/output.c.o
[ 38%] Building C object CMakeFiles/websockets_shared.dir/lib/parsers.c.o
[ 39%] Building C object CMakeFiles/websockets_shared.dir/lib/context.c.o
[ 40%] Building C object CMakeFiles/websockets_shared.dir/lib/alloc.c.o
[ 41%] Building C object CMakeFiles/websockets_shared.dir/lib/header.c.o
[ 43%] Building C object CMakeFiles/websockets_shared.dir/lib/client.c.o
[ 44%] Building C object CMakeFiles/websockets_shared.dir/lib/client-handshake.c.o
[ 45%] Building C object CMakeFiles/websockets_shared.dir/lib/client-parser.c.o
[ 46%] Building C object CMakeFiles/websockets_shared.dir/lib/ssl.c.o
[ 47%] Building C object CMakeFiles/websockets_shared.dir/lib/ssl-server.c.o
[ 48%] Building C object CMakeFiles/websockets_shared.dir/lib/ssl-client.c.o
[ 50%] Building C object CMakeFiles/websockets_shared.dir/lib/sha-1.c.o
[ 51%] Building C object CMakeFiles/websockets_shared.dir/lib/lws-plat-unix.c.o
[ 52%] Building C object CMakeFiles/websockets_shared.dir/lib/server.c.o
[ 53%] Building C object CMakeFiles/websockets_shared.dir/lib/server-handshake.c.o
[ 54%] Building C object CMakeFiles/websockets_shared.dir/lib/extension.c.o
[ 55%] Building C object CMakeFiles/websockets_shared.dir/lib/extension-permessage-deflate.c.o
[ 56%] Building C object CMakeFiles/websockets_shared.dir/lib/libev.c.o
[ 58%] Linking C shared library lib/libwebsockets.so
[ 58%] Built target websockets_shared
Scanning dependencies of target test-server
[ 59%] Building C object CMakeFiles/test-server.dir/test-server/test-server.c.o
[ 60%] Building C object CMakeFiles/test-server.dir/test-server/test-server-http.c.o
[ 61%] Building C object CMakeFiles/test-server.dir/test-server/test-server-dumb-increment.c.o
[ 62%] Building C object CMakeFiles/test-server.dir/test-server/test-server-mirror.c.o
[ 63%] Building C object CMakeFiles/test-server.dir/test-server/test-server-status.c.o
[ 65%] Building C object CMakeFiles/test-server.dir/test-server/test-server-echogen.c.o
[ 66%] Linking C executable bin/libwebsockets-test-server
[ 66%] Built target test-server
Scanning dependencies of target test-server-libev
[ 67%] Building C object CMakeFiles/test-server-libev.dir/test-server/test-server-libev.c.o
[ 68%] Building C object CMakeFiles/test-server-libev.dir/test-server/test-server-http.c.o
[ 69%] Building C object CMakeFiles/test-server-libev.dir/test-server/test-server-dumb-increment.c.o
[ 70%] Building C object CMakeFiles/test-server-libev.dir/test-server/test-server-mirror.c.o
[ 72%] Building C object CMakeFiles/test-server-libev.dir/test-server/test-server-status.c.o
[ 73%] Building C object CMakeFiles/test-server-libev.dir/test-server/test-server-echogen.c.o
[ 74%] Linking C executable bin/libwebsockets-test-server-libev
[ 74%] Built target test-server-libev
Scanning dependencies of target test-fraggle
[ 75%] Building C object CMakeFiles/test-fraggle.dir/test-server/test-fraggle.c.o
[ 76%] Linking C executable bin/libwebsockets-test-fraggle
[ 76%] Built target test-fraggle
Scanning dependencies of target test-echo
[ 77%] Building C object CMakeFiles/test-echo.dir/test-server/test-echo.c.o
[ 79%] Linking C executable bin/libwebsockets-test-echo
[ 79%] Built target test-echo
Scanning dependencies of target test-server-extpoll
[ 80%] Building C object CMakeFiles/test-server-extpoll.dir/test-server/test-server.c.o
[ 81%] Building C object CMakeFiles/test-server-extpoll.dir/test-server/test-server-http.c.o
[ 82%] Building C object CMakeFiles/test-server-extpoll.dir/test-server/test-server-dumb-increment.c.o
[ 83%] Building C object CMakeFiles/test-server-extpoll.dir/test-server/test-server-mirror.c.o
[ 84%] Building C object CMakeFiles/test-server-extpoll.dir/test-server/test-server-status.c.o
[ 86%] Building C object CMakeFiles/test-server-extpoll.dir/test-server/test-server-echogen.c.o
[ 87%] Linking C executable bin/libwebsockets-test-server-extpoll
[ 87%] Built target test-server-extpoll
Scanning dependencies of target test-ping
[ 88%] Building C object CMakeFiles/test-ping.dir/test-server/test-ping.c.o
[ 89%] Linking C executable bin/libwebsockets-test-ping
[ 89%] Built target test-ping
Scanning dependencies of target test-server-pthreads
[ 90%] Building C object CMakeFiles/test-server-pthreads.dir/test-server/test-server-pthreads.c.o
[ 91%] Building C object CMakeFiles/test-server-pthreads.dir/test-server/test-server-http.c.o
[ 93%] Building C object CMakeFiles/test-server-pthreads.dir/test-server/test-server-dumb-increment.c.o
[ 94%] Building C object CMakeFiles/test-server-pthreads.dir/test-server/test-server-mirror.c.o
[ 95%] Building C object CMakeFiles/test-server-pthreads.dir/test-server/test-server-status.c.o
[ 96%] Building C object CMakeFiles/test-server-pthreads.dir/test-server/test-server-echogen.c.o
[ 97%] Linking C executable bin/libwebsockets-test-server-pthreads
[ 97%] Built target test-server-pthreads
Scanning dependencies of target test-client
[ 98%] Building C object CMakeFiles/test-client.dir/test-server/test-client.c.o
[100%] Linking C executable bin/libwebsockets-test-client
[100%] Built target test-client
[ 27%] Built target websockets
[ 30%] Built target test-fuzxy
[ 58%] Built target websockets_shared
[ 66%] Built target test-server
[ 74%] Built target test-server-libev
[ 76%] Built target test-fraggle
[ 79%] Built target test-echo
[ 87%] Built target test-server-extpoll
[ 89%] Built target test-ping
[ 97%] Built target test-server-pthreads
[100%] Built target test-client
Install the project...
-- Install configuration: "Release"
-- Installing: /usr/local/lib/pkgconfig/libwebsockets.pc
-- Installing: /usr/local/lib/libwebsockets.a
-- Installing: /usr/local/include/libwebsockets.h
-- Installing: /usr/local/include/lws_config.h
-- Installing: /usr/local/lib/libwebsockets.so.8
-- Installing: /usr/local/lib/libwebsockets.so
-- Up-to-date: /usr/local/include/libwebsockets.h
-- Up-to-date: /usr/local/include/lws_config.h
-- Installing: /usr/local/bin/libwebsockets-test-client
-- Installing: /usr/local/bin/libwebsockets-test-server
-- Installing: /usr/local/bin/libwebsockets-test-fuzxy
-- Installing: /usr/local/bin/libwebsockets-test-server-pthreads
-- Installing: /usr/local/bin/libwebsockets-test-server-libev
-- Installing: /usr/local/bin/libwebsockets-test-server-extpoll
-- Up-to-date: /usr/local/bin/libwebsockets-test-client
-- Installing: /usr/local/bin/libwebsockets-test-fraggle
-- Installing: /usr/local/bin/libwebsockets-test-ping
-- Installing: /usr/local/bin/libwebsockets-test-echo
-- Up-to-date: /usr/local/share/libwebsockets-test-server/favicon.ico
-- Up-to-date: /usr/local/share/libwebsockets-test-server/leaf.jpg
-- Up-to-date: /usr/local/share/libwebsockets-test-server/libwebsockets.org-logo.png
-- Up-to-date: /usr/local/share/libwebsockets-test-server/test.html
-- Installing: /usr/local/share/libwebsockets-test-server/libwebsockets-test-server.key.pem
-- Installing: /usr/local/share/libwebsockets-test-server/libwebsockets-test-server.pem
-- Installing: /usr/local/lib/cmake/libwebsockets/LibwebsocketsConfig.cmake
-- Installing: /usr/local/lib/cmake/libwebsockets/LibwebsocketsConfigVersion.cmake
-- Old export file "/usr/local/lib/cmake/libwebsockets/LibwebsocketsTargets.cmake" will be replaced.  Removing files [/usr/local/lib/cmake/libwebsockets/LibwebsocketsTargets-release.cmake;/usr/local/lib/cmake/libwebsockets/LibwebsocketsTargets-debug.cmake].
-- Installing: /usr/local/lib/cmake/libwebsockets/LibwebsocketsTargets.cmake
-- Installing: /usr/local/lib/cmake/libwebsockets/LibwebsocketsTargets-release.cmake
```

One of the travis tests clones and builds it clean - every push - with that option as well, without problems.

As I say I can't reproduce it: if you can get another clue (since you're the only one with any symptom) I'll be happy to dig at it.  But as it is there's nothing do look at here, or at Travis.

**Comment by alexhultman** [comment]
My build is not including the libev.c:

`[ 26%] Building C object CMakeFiles/websockets.dir/lib/libev.c.o`

I'm on Fedora 22, I think you are either on Fedora 23 or 24. CMake is one big pile of garbage so it could be an issue with CMake and its libev plugin.

**Comment by lws-team** [comment]
I was on F22 until yesterday, I updated to F24 beta then.  But Travis isn't on either and he was happy before and now.

Cmake isn't perfect but before that we used autotools and that was really hell not just by itself, but when windows guys appeared sending patches for different incompatible msvc project files.  So I don't mind suffering a bit for cmake, which made all that go away.  But I doubt it's cmake since the only real confusion that has made for me is cached build options, and you're not doing that due to the clone.

One thought you are building in the tree, we recommend / document and only do / test building in a build dir, leaving the sources clean.  So you might try that.
